pkg/api/payer/nodeSelector.go

```diff
@@ -53,9 +54,19 @@ func (s *StableHashingNodeSelectorAlgorithm) GetNode(
 	spacing := maxHashSpace / numNodes
 
 	// Compute virtual positions for each node
+	// Skip nodes that are disabled or do not have an API enabled
 	nodeLocations := make([]uint32, numNodes)
-	for i := range nodes {
+	gotNodes := false
+	for i, node := range nodes {
+		if node.IsDisabled || !node.IsApiEnabled {
+			continue
+		}
 		nodeLocations[i] = uint32(i) * spacing
```

Comment:
I see your conversation with the AI below about the extra for loop, but I think having empty indexes in this array is going to be a footgun that will make the code harder to reason about. 

Maybe there is a third solution where we can use a single loop without having empty indexes in the `nodeLocations`?

Reply:
It shouldn't apply here in the payer. The contract registry gets rid of those empty nodes during `refreshData()`, right?
```
	newNodes := []Node{}
	for _, node := range fromContract {
		// nodes realistically start at 100, but the contract fills the array with empty nodes
		if !node.IsValidConfig {
			continue
		}
```

Reply:
Maybe we're taking about different things. We have a `nodeLocations` array of length `numNodes`, but only the indexes with `IsDisabled = false && IsApiEnabled = true` have data in them. The rest of the indexes are set to 0 because of the `continue` on line 62.

Reply:
Oh, I see. Good catch! I just modified a bit the test `TestGetNode_NoAvailableActiveNodes` and triggered an error because of this. Modifying this.

Reply:
Modified. I moved the banlist filter up as well, now we filter earlier, sort, and return the candidate.